src/main/java/io/cdap/plugin/dynamo/source/DynamoDBBatchSourceConfig.java

```diff
@@ -0,0 +1,316 @@
+    "(Macro Enabled)")
+  @Nullable
+  @Macro
+  private String nameMappings;
+
+  @Name(VALUE_MAPPINGS)
+  @Description("List of the placeholder tokens used as the attribute values in the 'Query or FilterQuery' along " +
+    "with their actual values. This is a comma-separated list of key-value pairs, where each pair is separated by a" +
+    " pipe sign '|' and specifies the tokens and actual values. For example, ':v_id|256,:v_title|B', if the query " +
+    "is like: 'ID = :v_id AND Title = :v_title'. (Macro Enabled)")
+  @Macro
+  private String valueMappings;
+
+  @Name(PLACEHOLDER_TYPE)
+  @Description("List of the placeholder tokens used as the attribute values in the 'Query or FilterQuery' along " +
+    "with their data types. This is a comma-separated list of key-value pairs, where each pair is separated by a " +
+    "pipe sign '|' and specifies the tokens and its type. For example, ':v_id|int,:v_title|string', if the query is" +
+    " like: 'ID = :v_id AND Title = :v_title'. Supported types are: 'boolean, int, long, float, double and string'." +
+    " (Macro Enabled)")
+  @Macro
+  private String placeholderType;
+
+  @Name(READ_THROUGHPUT)
+  @Description("Read Throughput for AWS DynamoDB table to connect to, in double. Default is 1. (Macro Enabled)")
+  @Nullable
+  @Macro
+  private String readThroughput;
+
+  @Name(READ_THROUGHPUT_PERCENTAGE)
+  @Description("Read Throughput Percentage for AWS DynamoDB table to connect to. Default is 0.5. (Macro Enabled)")
+  @Nullable
+  @Macro
+  private String readThroughputPercentage;
+
+  @Name(SCHEMA)
+  @Description("Specifies the schema that has to be output. If not specified, then by default each item will be " +
+    "emitted as a JSON string, in the 'body' field of the StructuredRecord.")
+  private String schema;
+
+  public DynamoDBBatchSourceConfig(String referenceName, String accessKey, String secretAccessKey,
+                                   @Nullable String region, @Nullable String endpointUrl, String tableName,
+                                   String query, @Nullable String filterQuery, @Nullable String nameMappings,
+                                   String valueMappings, String placeholderType, @Nullable String readThroughput,
+                                   @Nullable String readThroughputPercentage, String schema) {
+    super(referenceName);
+    this.accessKey = accessKey;
+    this.secretAccessKey = secretAccessKey;
+    this.region = region;
+    this.endpointUrl = endpointUrl;
+    this.tableName = tableName;
+    this.query = query;
+    this.filterQuery = filterQuery;
+    this.nameMappings = nameMappings;
+    this.valueMappings = valueMappings;
+    this.placeholderType = placeholderType;
+    this.readThroughput = readThroughput;
+    this.readThroughputPercentage = readThroughputPercentage;
+    this.schema = schema;
+  }
+
+  public String getAccessKey() {
+    return accessKey;
+  }
+
+  public void setAccessKey(String accessKey) {
```

Comment:
We try to make things immutable when possible and rarely have setters. This makes it much easier to reason about things, since we don't have to worry about things changing.

Everything should be set through the constructor, without these setters. Looking at the test cases where these things are used, I think what we need is a Builder with setters.